pkg/api/config/config.go

```diff
@@ -221,17 +222,32 @@ func SameFile(str1, str2 string) (bool, error) {
 	return os.SameFile(sFile, tFile), nil
 }
 
+func deepCopy(src, dst interface{}) error {
+	bytes, err := json.Marshal(src)
+	if err := inject.Error(err); err != nil {
+		return err
```

Comment:
It is strange that this line and the other check above are reported as not covered.

For the record I think `zotregistry.io/zot/pkg/test/inject` is one big hack and I would avoid importing test code from under `zotregistry.io/zot/pkg/test/` into production code.

Can you try to avoid using `zotregistry.io/zot/pkg/test/inject`? Can you try using a channel to produce errors? See https://github.com/project-zot/zot/blob/main/pkg/common/common_test.go#L34